### PR TITLE
FEAT: Wishlist Functionality

### PR DESCRIPTION
**Pull request by fahd23**
Added Wishlist Functionality
- Add to wishlist
- Remove from wishlist

### CODE REVIEW

File: src/components/product/Product.jsx [hunk 1]
```diff
@@ -1,216 +1,51 @@
...
-                  <BsSuitHeart />
+                <span
+                  className="favorite-btn"
+                  onClick={() => {
+                    wishlistDispatch({
+                      type: "ADD_TO_WISHLIST",
+                      payload: product,
+                    });
+                    console.log(product._id);
+                  }}
```
Comment (mariyasada) [inline src/components/product/Product.jsx]: remove console log

File: src/components/product/FilterOperations.jsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            <input
+              type="radio"
+              name="rate"
+              value="2"
+              checked={state.sortRating === "2"}
+              onClick={(e) =>
+                dispatch({ type: "RATING", payload: e.target.value })
+              }
+            />
+            <span>2 Stars and above </span>
```
Comment (JantuDeb) [inline src/components/product/FilterOperations.jsx]:
```
[4,3,2,1].map(rating=><div>)  you can  do something like this

```

**Review by mariyasada — APPROVED** [review]
the code structure is good. liked that you have created separate context for  cart and wishlist

File: src/components/product/FilterOperations.jsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+              className="check-input"
+              type="checkbox"
+              value="clothings"
+              checked={state.categoryName.clothings}
+              onClick={() => dispatch({ type: "CLOTHINGS" })}
+            />
+            <span>Clothings</span>
+          </label>
+        </div>
+      </div>
```
Comment (JantuDeb) [inline src/components/product/FilterOperations.jsx]: You can do the same here

**Review by JantuDeb — APPROVED** [review]
Overall the code looks fine

File: src/components/product/Product.jsx (hunk 1 above)
Comment (fahd23) [reply src/components/product/Product.jsx]: yeah, will remove it

File: src/components/product/FilterOperations.jsx (hunk 2 above)
Comment (fahd23) [reply src/components/product/FilterOperations.jsx]: will try to do that

File: src/components/product/FilterOperations.jsx (hunk 3 above)
Comment (fahd23) [reply src/components/product/FilterOperations.jsx]: Thank you so much for the suggestion

**Pull request merged**